python/pyspark/sql/pandas/serializers.py

```diff
@@ -388,13 +388,7 @@ def _create_batch(self, series):
 
         arrs = []
         for s, t in series:
-            if self._struct_in_pandas == "dict" and t is not None and pa.types.is_struct(t):
-                if not isinstance(s, pd.DataFrame):
-                    raise PySparkValueError(
-                        "A field of type StructType expects a pandas.DataFrame, "
-                        "but got: %s" % str(type(s))
-                    )
-
+            if t is not None and pa.types.is_struct(t) and isinstance(s, pd.DataFrame):
```

Comment:
cc @xinrong-meng @ueshin

Reply:
The changes cause a breaking change on Pandas UDF.

Pandas UDF should return `pd.DataFrame` when the return type is a struct type.
If it returns a `pd.Series`, it should fail with the error: `A field of type StructType expects a pandas.DataFrame`, but the changes make it allowed.

Reply:
Good to know. I've added a comment and a unit test to cover this case.